src/components/application_manager/src/resumption/resumption_data_db.cc

```diff
@@ -1006,6 +1016,59 @@ bool ResumptionDataDB::SelectSubscriptionsData(
   return true;
 }
 
+bool ResumptionDataDB::SelectUserLocationData(
+    const std::string& policy_app_id,
+    const std::string& device_id,
+    smart_objects::SmartObject& saved_app) const {
+  SDL_LOG_AUTO_TRACE();
+  using namespace app_mngr;
+  using namespace smart_objects;
+  uint32_t count_item = 0;
+  if (!SelectCountFromArray(
+          count_item, kSelectCountUserLocation, policy_app_id, device_id)) {
+    return false;
+  }
+
+  saved_app[strings::user_location] = SmartObject(SmartType_Map);
+
+  if (0 == count_item) {
+    SDL_LOG_INFO("Application does not contain user_location data");
+    return true;
+  }
+  utils::dbms::SQLQuery select_user_location(db());
+  if (!PrepareSelectQuery(select_user_location,
+                          policy_app_id,
+                          device_id,
+                          kSelectUserLocation)) {
+    SDL_LOG_WARN("Problem with verification user_location");
+    return false;
+  }
+  if (!select_user_location.Exec()) {
+    SDL_LOG_WARN("Problem with execution select_user_location query");
+    return false;
+  }
+  /* Position of data in "select_user_location" :
+     field "col"        from table "applicationUserLocation" = 0
+     field "colspan"    from table "applicationUserLocation" = 1
+     field "level"      from table "applicationUserLocation" = 2
+     field "levelspan"  from table "applicationUserLocation" = 3
+     field "row"        from table "applicationUserLocation" = 4
+     field "rowspan"    from table "applicationUserLocation" = 5*/
+  smart_objects::SmartObject grid =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  grid[rc_rpc_plugin::strings::kCol] = select_user_location.GetInteger(0);
+  grid[rc_rpc_plugin::strings::kColspan] = select_user_location.GetInteger(1);
+  grid[rc_rpc_plugin::strings::kLevel] = select_user_location.GetInteger(2);
+  grid[rc_rpc_plugin::strings::kLevelspan] = select_user_location.GetInteger(3);
+  grid[rc_rpc_plugin::strings::kRow] = select_user_location.GetInteger(4);
+  grid[rc_rpc_plugin::strings::kRowspan] = select_user_location.GetInteger(5);
+
+  saved_app[strings::user_location][rc_rpc_plugin::strings::kGrid] = grid;
+
+  SDL_LOG_INFO("UserLocation were restored from DB successfully");
```

Comment:
@VadymLuchko this log message is redundant. Prefer to log only error-return branches, but not a happy path

Reply:
fixed in bf0ce24216a70351394e14311bafcfac997606f6

Reply:
fixed in 40c8b7496a1967b7dacec511f2fd014a255fa36c